src/nvim/ui.c

```diff
@@ -448,3 +448,15 @@ Array ui_array(void)
   }
   return all_uis;
 }
+
+void ui_grid_resize(GridHandle grid_handle, int width, int height)
+{
+  ScreenGrid *grid = get_grid_by_handle(grid_handle);
+  if (grid == NULL) {
+    //TODO(utkarshme): error out
+    abort();
+    return;
+  }
+  grid_alloc(grid, height, width, true);
```

Comment:
We need to trigger a redraw also. If the grid belongs to a window, can use `redraw_win_later` (probably SOME_VALID, but not sure).

Actually we probably shouldn't reallocate here, redraw will always reallocate (and call `grid_resize` back)  if necessary anyway. Instead we shoud store the intended size so that futher redraws will respect it (and not overwrite it with the "outer" size), maybe in `internal_Rows`.